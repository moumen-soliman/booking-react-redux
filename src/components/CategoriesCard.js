import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class CategoriesCard extends Component {
    render() {
        const {id, name} = this.props
        return (
            <li><Link to={`/category/${id}`}>{name}</Link></li>
        )
    }
}

export default CategoriesCard;