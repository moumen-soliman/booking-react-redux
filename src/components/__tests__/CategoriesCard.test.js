import React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter} from 'react-router';
import CategoriesCard from '../CategoriesCard';

describe('Category Card', () => {
    test('should render props and links output', () => {
        const component = renderer.create(
            <MemoryRouter> 
                <CategoriesCard />
            </MemoryRouter>
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});