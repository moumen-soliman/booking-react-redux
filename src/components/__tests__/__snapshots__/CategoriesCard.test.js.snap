// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Card should render link output 1`] = `
<li>
  <a
    href="/category/undefined"
    onClick={[Function]}
  />
</li>
`;

exports[`Category Card should render props and links output 1`] = `
<li>
  <a
    href="/category/undefined"
    onClick={[Function]}
  />
</li>
`;
